From 521aab466e50f822298ae7bf817b14c1b9eaab74 Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Sun, 14 Mar 2021 03:20:43 +0200
Subject: [PATCH] Replace YYUSE with attribute unused in src/parser.y

The YYUSE macro was renamed to YY_USE in bison 3.7.5; might as well
avoid using it altogether, and instead mark the variable as unused.

Fixes the following linker error:

/usr/bin/ld: kafel/libkafel.a(libkafel.o): in function `kafel_yyerror':
arm_syscalls.c:(.text+0x6984): undefined reference to `YYUSE'
---
 src/parser.y | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/parser.y b/src/parser.y
index e0f109c..e5980c9 100644
--- a/src/parser.y
+++ b/src/parser.y
@@ -418,10 +418,9 @@ const_def
 
 %%
 
-void yyerror(YYLTYPE * loc, struct kafel_ctxt* ctxt, yyscan_t scanner,
-             const char *msg) {
+void yyerror(YYLTYPE * loc, struct kafel_ctxt* ctxt,
+             __attribute__((unused)) yyscan_t scanner, const char *msg) {
   if (!ctxt->lexical_error) {
-    YYUSE(scanner);
     if (loc->filename != NULL) {
       append_error(ctxt, "%s:%d:%d: %s", loc->filename, loc->first_line, loc->first_column, msg);
     } else {
